phpBB/phpbb/config/db.php

```diff
@@ -39,16 +42,28 @@ class db extends \phpbb\config\config
 	/**
 	* Creates a configuration container with a default set of values
 	*
-	* @param \phpbb\db\driver\driver_interface    $db    Database connection
-	* @param \phpbb\cache\driver\driver_interface $cache Cache instance
+	* @param db_interface    $db    Database connection
+	* @param cache_interface $cache Cache instance
 	* @param string                       $table Configuration table name
 	*/
-	public function __construct(\phpbb\db\driver\driver_interface $db, \phpbb\cache\driver\driver_interface $cache, $table)
+	public function __construct(db_interface $db, cache_interface $cache, $table)
 	{
 		$this->db = $db;
 		$this->cache = $cache;
 		$this->table = $table;
 
+		$this->initialise($cache);
+
+		parent::__construct($this->config);
+	}
+
+	/**
+	 * Initialise config with database and/or cached entries
+	 *
+	 * @param cache_interface $cache
+	 */
+	public function initialise(cache_interface $cache)
+	{
 		if (($config = $cache->get('config')) !== false)
 		{
 			$sql = 'SELECT config_name, config_value
```

Comment:
During installation process, `phpbb_config` table does not exist until some point, so this SQL query causes hidden (not reported) SQL error.
Probably needs a check for `_config` table existence and initializing `$config` as an empty array if the table doesn't exist.

Reply:
I take it this was already an issue before this PR then?

Reply:
Yeah, but looks like a good opportunity to fix it while changing the same code snippet :)

Reply:
Created a separate ticket for this: https://tracker.phpbb.com/browse/PHPBB3-16806